tests/ai/test_ai_manager.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021 RidgeRun, LLC (http://www.ridgerun.com)
+#  Authors: Daniel Chaves <daniel.chaves@ridgerun.com>
+#           Marisol Zeledon <marisol.zeledon@ridgerun.com>
+
+import cv2
+import numpy as np
+import random
+import unittest
+from unittest.mock import MagicMock
+
+from TI.postprocess import PostProcessDetection
+from TI.preprocess import PreProcessDetection
+from rr.ai.ai_manager import AIManager
+from rr.ai.ai_manager import AIManagerError
+from rr.ai.ai_manager import AIManagerOnNewImage
+
+model = "/opt/edge_ai_apps/models/detection/TFL-OD-200-ssd-mobV1-coco-mlperf-300x300/"
+width = 1920
+height = 1080
+color = (100, 100, 100)
+disp_width = 2040
+disp_height = 1920
+
+
+class MockImage():
+    def __init__(self, width, height, color):
+        self.mock_img = self.create_img(width, height, rgb_color=color)
+
+    def create_img(self, width, height, rgb_color=(0, 0, 0)):
+        img = np.zeros((height, width, 3), np.uint8)
+
+        color = tuple(reversed(rgb_color))
+        img[:] = color
+
+        return img
+
+
+class TestAIManager(unittest.TestCase):
+    def setUp(self):
+        global model, width, height, disp_width, disp_height
+        global width, height, color
+
+        self.model = model
+        self.img = MockImage(width, height, color).mock_img
```

Comment:
You pass in the class, the class has a `get_data` method that returns the actual data.